Islands.java

```diff
@@ -0,0 +1,70 @@
+/**
+* You are given a 2-dimensional map of tiles. 
+* Each tile is either land or water. 
+* You have to write a function that counts the number of islands.
+*
+* @author  Argyro Gounari
+* @version 1.0
+* @since   01-07-2018 
+*/
+public class Islands {
+
+	/**
+	 * Function that counts the number of islands.
+	 * False -> Water
+	 * True -> Land
+	*/
+	public int islandsCount(Boolean[][] map) {
+		
+		int count = 0;
+		for(int i=0; i<map.length; i++) {
+	        for(int j=0; j<map[i].length; j++) {
+	        	if (map[i][j] == true) {
+					count++;
+	        	}
+	        	map = sameIsland(map, i, j);
+	        }
+	    }
+		
+		return count;
+		
+	}
+	
+	/**
+	 * Two tiles belong to the same island 
+	 * if they are both land and are adjacent 
+	 * horizontally or vertically, but not diagonally.
+	*/
+	private Boolean[][] sameIsland(Boolean[][] map, int collum, int row) {
+		if((row<0 || collum<0 || collum>=map.length || row>=map[collum].length) || map[collum][row]==false) 
```

Comment:
You use this line of code more than once. Why not extract it into a helper method.